### PR TITLE
test that sqlite library is compiled with threadsafe code

### PR DESCRIPTION
**Pull request by efim-poberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Agent/Store/SQLite.hs [hunk 1]
```diff
@@ -55,7 +57,12 @@ createSQLiteStore dbFilename = do
 connectSQLiteStore :: MonadUnliftIO m => String -> m SQLiteStore
 connectSQLiteStore dbFilename = do
   dbConn <- liftIO $ DB.open dbFilename
-  return SQLiteStore {dbFilename, dbConn}
+  compileOptions <- liftIO (DB.query_ dbConn "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
+  let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
+  case threadsafeOption of
+    Just "THREADSAFE=0" -> error "SQLite compiled with not threadsafe code" -- TODO warn instead
+    Just _ -> return SQLiteStore {dbFilename, dbConn} -- TODO info threadsafe
+    Nothing -> error "SQLite THREADSAFE compile option not found" -- should not happen
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent/Store/SQLite.hs:65]: Better put it into createSQLiteStore which is called once, connectSQLiteStore is called once per thread (twice for each TCP connection)

Comment (epoberezkin) [reply src/Simplex/Messaging/Agent/Store/SQLite.hs:65]: If option not found - warn and continue
If option found and incorrect - we could exit or ask y/n to continue

File: src/Simplex/Messaging/Agent/Store/SQLite.hs [hunk 2]
```diff
@@ -55,7 +57,12 @@ createSQLiteStore dbFilename = do
 connectSQLiteStore :: MonadUnliftIO m => String -> m SQLiteStore
 connectSQLiteStore dbFilename = do
   dbConn <- liftIO $ DB.open dbFilename
-  return SQLiteStore {dbFilename, dbConn}
+  compileOptions <- liftIO (DB.query_ dbConn "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
+  let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent/Store/SQLite.hs:61]: what does it actually return - a single item result that is array of texts? Just was wondering if it should be just maybeHead or something...

File: tests/AgentTests/SQLiteTests.hs [hunk 3]
```diff
@@ -71,6 +73,12 @@ storeTests = withStore do
       describe "SndQueue exists" testCreateSndMsg
       describe "SndQueue doesn't exist" testCreateSndMsgNoQueue
 
+testCompiledThreadsafe :: SpecWith SQLiteStore
+testCompiledThreadsafe = do
+  it "should throw error if compiled sqlite library is not threadsafe" $ \store -> do
+    compileOptions <- DB.query_ (dbConn store) "pragma COMPILE_OPTIONS;" :: IO [[T.Text]]
+    shouldNotContain compileOptions [["THREADSAFE=0"]]
```
Comment (epoberezkin) [inline tests/AgentTests/SQLiteTests.hs:80]: better to write it as ```compileOptions `shouldNotContain` [["THREADSAFE=0"]]```

Comment (epoberezkin) [reply tests/AgentTests/SQLiteTests.hs:80]: but I don't think it is correct - try testing for what it definitely contains

File: src/Simplex/Messaging/Agent/Store/SQLite.hs (hunk 2 above)
Comment (efim-poberezkin) [reply src/Simplex/Messaging/Agent/Store/SQLite.hs:61]: array of arrays actually, that's why I do `concat`

File: tests/AgentTests/SQLiteTests.hs (hunk 3 above)
Comment (efim-poberezkin) [reply tests/AgentTests/SQLiteTests.hs:80]: yep, right.. I was trying `compileOptions shouldNotContain [["THREADSAFE=0"]]` but missed backtics. But it is correct, I checked

File: src/Simplex/Messaging/Agent/Store/SQLite.hs (hunk 1 above)
Comment (efim-poberezkin) [reply src/Simplex/Messaging/Agent/Store/SQLite.hs:65]: connectSQLiteStore +

I remember why I put it in createSQLiteStore - I was trying to be safe - but after giving it a second thought I see my reasoning was incorrect.

nvm this, I'll fix it

**Review by epoberezkin — APPROVED** [review]

**Pull request merged**